File: 169-majority-element/majority-element.java
class Solution {
    public int majorityElement(int[] nums) {
        int n=nums.length/2;
      
    Arrays.sort(nums);
    int count=0;
    if(nums.length==1)
    {
        return nums[0];
    }

    for(int i=1;i<nums.length;i++)
    {
        if(nums[i-1]==nums[i])
        {
        count++;
        }
        else
        {
            count=0;
        }
        if((count+1)>n)
        {
            return nums[i];
        }
    }
    return -1;

    }
}

File: 169-majority-element/majority-element.cpp
class Solution {
public:
    int majorityElement(vector<int>& nums) {
        map<int,int> freq;
        int ans=0;
        int n= nums.size();
        for(int num:nums){
            freq[num]++;
        }
        for(auto it:freq){
            if(it.second>n/2) ans=it.first;
        }
        return ans;
    }
};